code/game/mecha/equipment/tools/work_tools.dm

```diff
@@ -261,6 +261,13 @@
 					S.ChangeTurf(/turf/simulated/floor/plating)
 					playsound(S, usesound, 50, 1)
 					chassis.spark_system.start()
+			if(ischasm(target))
+				var/turf/simulated/floor/chasm/C = target
+				occupant_message("Building Floor...")
+				if(do_after_cooldown(C))
+					C.ChangeTurf(/turf/simulated/floor/plating)
+					playsound(C, usesound, 50, 1)
+					chassis.spark_system.start()
```

Comment:
Couldn't you do `if(isspaceturf(target) || ischasm(target))` on line 257?

Reply:
not thinking about it, i`l do it soon